### ISSUE TITLE
Short sound effect file with brown noise sounds distorted

### ISSUE THREAD

**Issue by janstadt** [music-assistant/hass-music-assistant#2059]
### What version of Music Assistant has the issue?

Latest beta

### What version of the Home Assistant Integration have you got installed?

Latest beta

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

I have a simple wav file that i loop over ighr for white noise. It plays flawlessly through LMS (squeezelite player) thats installed on a picoreplayer of mine but barely plays through MA.   Ive tried volume normalization on/off, changed tje negative number, tried different file extensions, tried converting it to mp3 but nothing seems to play well when i play it theough an automation. 

### How to reproduce

Have an automation set volume, repeat, and start playing a file. Notice that there is clipping, dostortion and eventually the file stops playing. 

### Music Providers

Local HA file, but i also have spotify integration. 

### Player Providers

Squeezelite. 

### What version of Home Assistant Core are your running

2024.3.1

### What type of installation are you running?

Home Assistant Container

### On what type of hardware are you running?

Linux

**Comment by OzGav** [comment]
We will need the full log attached

**Comment by janstadt** [comment]
Thanks @OzGav. You're talking about MA logs and not HA logs correct? I am anonymizing them currently. Is there a way to do that automatically? That would be a nice feature for debugging purposes. By the looks of the errors in the logs, im getting a bunch of authz errors. This is my automation. Keep in mind i also tried different files so the one in the automation is different than the one in the MA logs. Curious if i should be trying to use the MA service instead of the built in HA media_player services? Everything works perfectly with regular old LMS. Would love the UI improvements of MA vs LMS. 

```
alias: Play White Noise In Noahs Room In the Morning
description: ""
trigger:
  - platform: time
    at: "20:00:00"
  - platform: time
    at: "09:00:00"
    id: "off"
condition: []
action:
  - if:
      - condition: trigger
        id:
          - "off"
    then:
      - service: media_player.media_pause
        target:
          entity_id: []
          area_id: []
          device_id:
            - 567aab518b3f8031517ba1e052ac1bff
            - cd63028b604a32be5410ba488f6111e9
        data: {}
      - service: media_player.repeat_set
        metadata: {}
        data:
          repeat: "off"
        target:
          device_id:
            - 567aab518b3f8031517ba1e052ac1bff
            - cd63028b604a32be5410ba488f6111e9
    else:
      - service: media_player.volume_set
        data:
          volume_level: 0.39
        target:
          device_id:
            - cd63028b604a32be5410ba488f6111e9
            - 567aab518b3f8031517ba1e052ac1bff
      - service: media_player.repeat_set
        metadata: {}
        data:
          repeat: all
        target:
          device_id:
            - 567aab518b3f8031517ba1e052ac1bff
            - cd63028b604a32be5410ba488f6111e9
      - service: media_player.play_media
        target:
          entity_id: media_player.gretas_room
        data:
          media_content_id: media-source://media_source/local/brown-noise.wav
          media_content_type: audio/x-wav
        metadata:
          title: brown-noise.wav
          thumbnail: null
          media_class: music
          children_media_class: null
          navigateIds:
            - {}
            - media_content_type: app
              media_content_id: media-source://media_source
      - service: media_player.play_media
        target:
          entity_id: media_player.noahs_room
        data:
          media_content_id: media-source://media_source/local/brown-noise.wav
          media_content_type: audio/x-wav
        metadata:
          title: brown-noise.wav
          thumbnail: null
          media_class: music
          children_media_class: null
          navigateIds:
            - {}
            - media_content_type: app
              media_content_id: media-source://media_source
mode: single

```

[musicassistant.log](https://github.com/music-assistant/hass-music-assistant/files/14744279/musicassistant.log)
[musicassistant.1.log](https://github.com/music-assistant/hass-music-assistant/files/14744282/musicassistant.1.log)

**Comment by OzGav** [comment]
Any reason why you are using device_id? The docs say to use entity_id https://www.home-assistant.io/integrations/media_player/

I see the authorisation error. The question is why are you trying to send a random wave file to a MA player? If you want to just play the track then just send it to the HA player. If you want to use MA then the file needs to be in the MA library. In order for that to happen you will need to add some basic tags to it, name, album, artist, album artist. You can just use Various Artists for the last two if you wish. Just make up an album name.

**Comment by janstadt** [comment]
I currently have spotify integrated and plex for local music. I can try and add the files to plex and access them through MA that way. Just odd that it works flawlessly in LMS.

**Comment by OzGav** [comment]
You can’t have the LMS server running at the same time as MA. So disable LMS

**Comment by marcelveldt** [comment]
There is a possibility that this is caused by sending a wave file from HA, that scenario was never tested (as normally you just play music from the MA library or an incidental TTS mp3). We'll have to test this scenario where the WAV files are stored in HA.

**Comment by marcelveldt** [comment]
> You can’t have the LMS server running at the same time as MA. So disable LMS

You can, as long as you dont use the slimproto provider in MA. Both servers cant be active at the same time

**Comment by janstadt** [comment]
Yep, ive never ran the at the same time. I was just explaining how i did things while each backend was running. I also tried with an mp3 file and while it worked better, i still had issues with sound cutting/clipping/stopping.

**Comment by OzGav** [comment]
Try importing the file into the MA library (please note the tagging I mentioned before) and then try playing it from the UI

**Comment by marcelveldt** [comment]
You mention the sound clipping/distorted ? How are your volume normalization settings setup ?
Also, can you maybe send me one of those files so I can try to reproduce it here ?

BTW: Please make sure to run on the latest beta 124, the betas 120..122 were flawed with a performance issue

**Comment by janstadt** [comment]
I added it to my plex server and its playing better. What im trying to replicate now is to play this track on repeat, but when using the MA service to play the track it disregards the call to set the volume and repeat on the devices. It just plays the single track and completes. I also tried putting the `mass.play_media` call as the last action but that didnt matter either. I'd really prefer not to make a playlist of 12hrs of the 9 second wav file. I think the repeat and volume set should be honored. I can try `device_id` for as opposed to `entity_id` and will report back but hopefully someone has a solution. I tried grabbing random brown noise playlists on spotify but they all have random intros that are annoying as hell.

HA Automation snippet:

```
else:
  - service: mass.play_media
    metadata: {}
    data:
      media_type: track
      media_id:
        - Brown Noise
      enqueue: play
      artist: Various Artists
      album: Various Artists
    target:
      entity_id:
        - media_player.gretas_room
        - media_player.noahs_room
  - service: media_player.volume_set
    data:
      volume_level: 0.39
    target:
      entity_id:
        - media_player.gretas_room
        - media_player.noahs_room
  - service: media_player.repeat_set
    metadata: {}
    data:
      repeat: all
    target:
      entity_id:
        - media_player.gretas_room
        - media_player.noahs_room
```

**Comment by janstadt** [comment]
Oh sweet, `device_id` is honored and its repeating.

**Comment by janstadt** [comment]
Another update. I tried adding the file to the local file system music provider and it plays once but replay doesnt seem to work like it did with plex. I'll try with `entity_id` and see if that works (`device_id` seemed to work with plex).

```
2024-03-26 19:27:58.963 INFO (MainThread) [music_assistant.music] Sync task for Filesystem (local disk) completed
2024-03-26 19:35:48.533 ERROR (MainThread) [music_assistant.webserver] Error handling message: players/queue/play_pause: Resume queue requested but queue d8:3a:dd:ba:cd:16 is empty

```

I cant imagine creating a playlist or enqueuing thousands of the same file is gonna be super performant.

**Comment by janstadt** [comment]
device vs entity didnt solve this. Very odd i was able to get it working with plex as a media source but not local file. 

Does this look appropriate:

```
service: mass.play_media
target:
  device_id:
    - 8050982310d7bc8602d6218484758b15
    - 474a13f1497e8f1bdc483e84c5485c1e
data:
  media_id: 10795
  media_type: track
```

It would be great if we could add `repeat` and other things to the data section to tell MA to repeat like the Media Player integration does in HA.

**Comment by janstadt** [comment]
So if i select a the file in MA and select "Play Now", and then set the repeat button in the player it doesnt repeat the file so i think my HA automations are correct. Any ideas why the repeat functionality wouldnt work?

**Comment by janstadt** [comment]
Could this be related to https://github.com/music-assistant/hass-music-assistant/issues/1967 ? I know @OzGav mentioned he cant repro on slimproto, but thats what im using and i cannot for the life of me get a file to repeat either via the HA integration or via MA directly. 

I bumped down the logging to verbose to see if i can catch anything odd happening:

```
2024-03-26 20:52:28.122 DEBUG (MainThread) [music_assistant.audio] Using cached streamdetails for filesystem_local--MHnmpHLE/brown-noise.mp3
2024-03-26 20:52:28.123 DEBUG (MainThread) [music_assistant.audio] Using cached streamdetails for filesystem_local--MHnmpHLE/brown-noise.mp3
2024-03-26 20:52:28.134 DEBUG (MainThread) [music_assistant.players] Handling command cmd_volume_set for player Noahs Room
2024-03-26 20:52:28.134 DEBUG (MainThread) [music_assistant.players] Handling command cmd_volume_set for player Gretas Room
2024-03-26 20:52:28.281 DEBUG (MainThread) [music_assistant.streams] Got GET request to /single/d8:3a:dd:ba:cd:16/ed4400f08d9e47e491a2926d0942e387.flac from 192.168.1.209
headers: <CIMultiDictProxy('Host': '192.168.1.68:8098', 'Connection': 'close', 'Accept': '*/*', 'Cache-Control': 'no-cache', 'User-Agent': 'VLC/3.0.9 LibVLC/3.0.9', 'Range': 'bytes=0-')>

2024-03-26 20:52:28.281 DEBUG (MainThread) [music_assistant.streams] Start serving audio stream for QueueItem library://track/1 to Noahs Room
2024-03-26 20:52:28.283 DEBUG (MainThread) [music_assistant.streams.media_stream] start media stream for: filesystem_local--MHnmpHLE://track/brown-noise.mp3
2024-03-26 20:52:28.334 DEBUG (MainThread) [music_assistant.streams] Got GET request to /single/d8:3a:dd:ba:cd:d1/386a293a01e9497cb6588b1a9fa3936c.flac from 192.168.1.222
headers: <CIMultiDictProxy('Host': '192.168.1.68:8098', 'Connection': 'close', 'Accept': '*/*', 'Cache-Control': 'no-cache', 'User-Agent': 'VLC/3.0.9 LibVLC/3.0.9', 'Range': 'bytes=0-')>

2024-03-26 20:52:28.334 DEBUG (MainThread) [music_assistant.streams] Start serving audio stream for QueueItem library://track/1 to Gretas Room
2024-03-26 20:52:28.337 DEBUG (MainThread) [music_assistant.streams.media_stream] start media stream for: filesystem_local--MHnmpHLE://track/brown-noise.mp3
2024-03-26 20:52:28.466 VERBOSE (MainThread) [music_assistant.streams] Input #0, mp3, from '/data/media/brown-noise.mp3':
2024-03-26 20:52:28.466 VERBOSE (MainThread) [music_assistant.streams] Metadata:
2024-03-26 20:52:28.466 VERBOSE (MainThread) [music_assistant.streams] title           : Brown Noise
2024-03-26 20:52:28.466 VERBOSE (MainThread) [music_assistant.streams] album           : Varioius Artists
2024-03-26 20:52:28.466 VERBOSE (MainThread) [music_assistant.streams] album_artist    : Various Artists
2024-03-26 20:52:28.466 VERBOSE (MainThread) [music_assistant.streams] track           : 1
2024-03-26 20:52:28.466 VERBOSE (MainThread) [music_assistant.streams] disc            : 1
2024-03-26 20:52:28.466 VERBOSE (MainThread) [music_assistant.streams] artist          : Various Artists
2024-03-26 20:52:28.466 VERBOSE (MainThread) [music_assistant.streams] lyrics-eng      :
2024-03-26 20:52:28.466 VERBOSE (MainThread) [music_assistant.streams] date            : 2024
2024-03-26 20:52:28.467 VERBOSE (MainThread) [music_assistant.streams] Duration: 00:00:09.55, start: 0.023021, bitrate: 67 kb/s
2024-03-26 20:52:28.467 VERBOSE (MainThread) [music_assistant.streams] Stream #0:0: Audio: mp3, 48000 Hz, mono, fltp, 64 kb/s
2024-03-26 20:52:28.467 VERBOSE (MainThread) [music_assistant.streams] Stream mapping:
2024-03-26 20:52:28.467 VERBOSE (MainThread) [music_assistant.streams] Stream #0:0 -> #0:0 (mp3 (mp3float) -> pcm_s16le (native))
2024-03-26 20:52:28.467 VERBOSE (MainThread) [music_assistant.streams] Press [q] to stop, [?] for help
2024-03-26 20:52:28.473 VERBOSE (MainThread) [music_assistant.streams] Output #0, s16le, to 'pipe:':
2024-03-26 20:52:28.473 VERBOSE (MainThread) [music_assistant.streams] Metadata:
2024-03-26 20:52:28.473 VERBOSE (MainThread) [music_assistant.streams] title           : Brown Noise
2024-03-26 20:52:28.473 VERBOSE (MainThread) [music_assistant.streams] album           : Varioius Artists
2024-03-26 20:52:28.473 VERBOSE (MainThread) [music_assistant.streams] album_artist    : Various Artists
2024-03-26 20:52:28.473 VERBOSE (MainThread) [music_assistant.streams] track           : 1
2024-03-26 20:52:28.473 VERBOSE (MainThread) [music_assistant.streams] disc            : 1
2024-03-26 20:52:28.473 VERBOSE (MainThread) [music_assistant.streams] artist          : Various Artists
2024-03-26 20:52:28.473 VERBOSE (MainThread) [music_assistant.streams] lyrics-eng      :
2024-03-26 20:52:28.473 VERBOSE (MainThread) [music_assistant.streams] date            : 2024
2024-03-26 20:52:28.473 VERBOSE (MainThread) [music_assistant.streams] encoder         : Lavf60.3.100
2024-03-26 20:52:28.474 VERBOSE (MainThread) [music_assistant.streams] Stream #0:0: Audio: pcm_s16le, 48000 Hz, stereo, s16, 1536 kb/s
2024-03-26 20:52:28.474 VERBOSE (MainThread) [music_assistant.streams] Metadata:
2024-03-26 20:52:28.474 VERBOSE (MainThread) [music_assistant.streams] encoder         : Lavc60.3.100 pcm_s16le
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] Input #0, mp3, from '/data/media/brown-noise.mp3':
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] Metadata:
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] title           : Brown Noise
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] album           : Varioius Artists
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] album_artist    : Various Artists
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] track           : 1
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] disc            : 1
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] artist          : Various Artists
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] lyrics-eng      :
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] date            : 2024
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] Duration: 00:00:09.55, start: 0.023021, bitrate: 67 kb/s
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] Stream #0:0: Audio: mp3, 48000 Hz, mono, fltp, 64 kb/s
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] Stream mapping:
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] Stream #0:0 -> #0:0 (mp3 (mp3float) -> pcm_s16le (native))
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] Press [q] to stop, [?] for help
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] Output #0, s16le, to 'pipe:':
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] Metadata:
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] title           : Brown Noise
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] album           : Varioius Artists
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] album_artist    : Various Artists
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] track           : 1
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] disc            : 1
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] artist          : Various Artists
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] lyrics-eng      :
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] date            : 2024
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] encoder         : Lavf60.3.100
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] Stream #0:0: Audio: pcm_s16le, 48000 Hz, stereo, s16, 1536 kb/s
2024-03-26 20:52:28.579 VERBOSE (MainThread) [music_assistant.streams] Metadata:
2024-03-26 20:52:28.580 VERBOSE (MainThread) [music_assistant.streams] encoder         : Lavc60.3.100 pcm_s16le
2024-03-26 20:52:28.770 VERBOSE (MainThread) [music_assistant.streams] size=       0kB time=-577014:32:22.77 bitrate=  -0.0kbits/s speed=N/A    
size=    1783kB time=00:00:09.59 bitrate=1521.7kbits/s speed=31.6x
2024-03-26 20:52:28.771 VERBOSE (MainThread) [music_assistant.streams] video:0kB audio:1783kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.000000%
2024-03-26 20:52:28.777 VERBOSE (MainThread) [music_assistant.streams] size=       0kB time=-577014:32:22.77 bitrate=  -0.0kbits/s speed=N/A    
size=    1783kB time=00:00:09.59 bitrate=1521.7kbits/s speed=36.1x
2024-03-26 20:52:28.777 VERBOSE (MainThread) [music_assistant.streams] video:0kB audio:1783kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.000000%
2024-03-26 20:52:28.790 DEBUG (MainThread) [music_assistant.streams.media_stream] stream finished for: filesystem_local--MHnmpHLE://track/brown-noise.mp3 (9.510479166666666 seconds streamed, exitcode 0)
2024-03-26 20:52:28.791 DEBUG (MainThread) [music_assistant.streams.media_stream] Loudness measurement for filesystem_local--MHnmpHLE://track/brown-noise.mp3: LoudnessMeasurement(integrated=-25.24, true_peak=-1.61, lra=0.3, threshold=-35.24)
2024-03-26 20:52:28.795 DEBUG (MainThread) [music_assistant.streams.media_stream] stream aborted for: filesystem_local--MHnmpHLE://track/brown-noise.mp3 (8.0 seconds streamed, exitcode None)
2024-03-26 20:52:30.302 DEBUG (MainThread) [music_assistant.audio] Using (pre)cached streamdetails from queue_item for library://track/1
2024-03-26 20:52:30.302 DEBUG (MainThread) [music_assistant.audio] Using (pre)cached streamdetails from queue_item for library://track/1
2024-03-26 20:52:30.619 DEBUG (MainThread) [music_assistant.audio] Using (pre)cached streamdetails from queue_item for library://track/1
2024-03-26 20:52:30.620 DEBUG (MainThread) [music_assistant.audio] Using (pre)cached streamdetails from queue_item for library://track/1
```

**Comment by marcelveldt** [comment]
That issue is unrelated.
repeat works for everyone using MA. 

Are you 100% sure you are using the entity id's of the MA players ?
Also, you dont need to set repeat every time. Its a setting of the queue attached to a player. So if enabled, it will do repeat until you disable it again.

Maybe there is some regression in the last couple of betas. We had a fair share of annoying issues in the betas 129..124.
The latest beta (125) should be pretty solid again. I'll do a quick test of repeat on slimproto tomorrow but I'm 99% that works as expected.

I would add the files to the local provider btw in MA, there really is no advantage of plex in between.

**Comment by janstadt** [comment]
Yeah after i added it back to plex the repeat isnt working any longer. Ill add it back to the local provider and test some more. The point about repeat being set once is good to know. I can try and remove that. I would assume if you try and set it again that it should just stick or be ignored if already set. If you were to set up an HA automation for a single song to be played on repeat, would it look any different than what i have pasted in the few posts above? Ive wiped my MA config directory a couple times now and started fresh but it doesnt seem to change anything. Was hoping i had some corrupted data causing repeat to fail.

**Comment by OzGav** [comment]
What happens in the queue? Does the track moved to played items? Is it an mp3? If not can you convert it to mp3 and try again?This works perfectly for me on a picoreplayer

**Comment by janstadt** [comment]
Its an mp3. Here is a clip of the playlist after i add the file to the playlist. 

https://github.com/music-assistant/hass-music-assistant/assets/1872087/d3f47346-bb68-4d70-ae46-d9527e272e8b

**Comment by OzGav** [comment]
Thats interesting as it behaves as if the setting is off. Have you tried all your other players to see what happens with them?

**Comment by janstadt** [comment]
Yep. Same happens with the two other players in my kids rooms.

**Comment by OzGav** [comment]
And what sort of players are those?

**Comment by janstadt** [comment]
they're all picoreplayers. piCorePlayer v8.2.0 | www v00017 | linux 5.15.35-pcpCore-v7 (32) | piCore v13.2 | Squeezelite v1.9.9-1392-pCP 

Keep in mind, these devices all play as expected when using LMS installed on a pi. I dont change the picoreplayers at all, just switch between MA and LMS. When i turn LMS off and reconfigure MA is the only time i cant get these things to repeat as expected. 

Here is a config for the player. 

```
# For more options and information see
# http://rpf.io/configtxt
# Some settings may impact device functionality. See link above for details

[PI0]
initramfs rootfs-pCP-13.2.gz,modules-5.15.35-pcpCore.gz followkernel
kernel kernel51535.img

[PI02]
initramfs rootfs-pCP-13.2.gz,modules-5.15.35-pcpCore-v7.gz followkernel
kernel kernel51535v7.img

[PI1]
initramfs rootfs-pCP-13.2.gz,modules-5.15.35-pcpCore.gz followkernel
kernel kernel51535.img

[PI2]
initramfs rootfs-pCP-13.2.gz,modules-5.15.35-pcpCore-v7.gz followkernel
kernel kernel51535v7.img

[PI3]
initramfs rootfs-pCP-13.2.gz,modules-5.15.35-pcpCore-v7.gz followkernel
kernel kernel51535v7.img

[PI4]
initramfs rootfs-pCP-13.2.gz,modules-5.15.35-pcpCore-v7l.gz followkernel
kernel kernel51535v7l.img

[CM4]
otg_mode=1

[ALL]

cmdline cmdline.txt

# uncomment if you get no picture on HDMI for a default "safe" mode
#hdmi_safe=1

# uncomment this if your display has a black border of unused pixels visible
# and your display can output without overscan
#disable_overscan=1

# uncomment the following to adjust overscan. Use positive numbers if console
# goes off screen, and negative if there is too much border
#overscan_left=16
#overscan_right=16
#overscan_top=16
#overscan_bottom=16

# uncomment to force a console size. By default it will be display's size minus
# overscan.
#framebuffer_width=1280
#framebuffer_height=720

# uncomment if hdmi display is not detected and composite is being output

# uncomment to force a specific HDMI mode (this will force VGA)
#hdmi_group=1
#hdmi_mode=1

# uncomment to force a HDMI mode rather than DVI. This can make audio work in
# DMT (computer monitor) modes

# uncomment to increase signal to HDMI, if you have interference, blanking, or
# no display
#config_hdmi_boost=4

# uncomment for composite PAL
#sdtv_mode=2

#----------------------------------------------------
dtparam=i2c_arm=on,i2s=on

#---pCP----------------------------------------------
# pCP does not use graphics chip by default, free up most memory
#gpu_mem=16
initial_turbo=60

# onboard audio overlay
dtparam=audio=on
audio_pwm_mode=2

# For Jivelite use with HDMI screens
# Comment out the lines above
#   gpu_mem=xx
#   dtparam=audio=on
# Then uncomment the following line.
#dtoverlay=vc4-kms-v3d
#

#Disable Bluetooth on both rpi3 and rpi0W
dtoverlay=disable-bt

# uncomment to overclock the arm. 700 MHz is the default.
#arm_freq=
#gpu_freq=
#core_freq=
#sdram_freq=
#over_voltage=
force_turbo=1

# Remove audio hiss
# disable_audio_dither=1

#If booting from USB or network, enable this option to reduce CPU load looking for a non-existing card.
#dtparam=sd_poll_once

# Screen rotation 0 => 0 degrees, 2 => 180 degrees
lcd_rotate=2

#Custom Configuration Area, for config settings that are not managed by pCP.
#pCP will retain these settings during insitu-update
#---Begin-Custom-(Do not alter Begin or End Tags)-----
#---End-Custom----------------------------------------

```

**Comment by OzGav** [comment]
Its a tough one as I have excactly the same setup and mine works fine except I don't have LMS on the network. ALso I have not changed anything in the PCP settings so if you have then I would look carefully at those. If you have the option perhaps try a fresh install of PCP on one of the devices, do nothing with it except have it discovered by MA and then try it?

**Comment by janstadt** [comment]
Again, I only run MA or LMS at a time, never both. I have updated some settings according to some post i found. Mostly around performance like pinning cores and force turbo. Its odd that it plays fine with LMS though but not MA.

**Comment by janstadt** [comment]
I did a clean flash on one of my players and its still just playing once. I did notice that there is a new version of pcp that released this past week. I wonder if anyone has been running that new version and whether or not it has a fix or something in it? The fact that it doesnt exhibit this behavior via lms directly though and only through MA makes me think somethings going on with MA still.

**Comment by marcelveldt** [comment]
I think this is a bug on our side with repeat of the full queue. I think/hope that single track repeat will work.

I will reproduce it here and fix.
I'll let you know here when there's progress

**Comment by OzGav** [comment]
Please keep an eye out for beta 134 and test again and report back once that is released.

**Comment by OzGav** [comment]
@janstadt Please advise if this is still a problem for you in beta 134

**Comment by janstadt** [comment]
Pulled latest with a clean install (no configs/providers/players configured). This was installed the same way as i used to install it:

```
  mass:
    image: ghcr.io/music-assistant/server
    container_name: music-assistant
    network_mode: host
    restart: unless-stopped
    privileged: true
    ports:
      - 8095:8095
    volumes:
      - ${ROOT}/config/music-assistant:/data
```

When i run it like this, i cannot add the slimproto provider as it says that a port is already in use. I look at my ports and its indeed being used, but its tied back to music assistant. If I take down the container the ports are no longer in use (9000 and 3483). If i start the container up without a network_mode im able to add slimproto, but the IP that gets configured in MA is the docker container IP and not my host IP. I can change the webserver address, but the slimproto service doesnt discover any speakers. Has anything changed with this kinda behavior? I cant figure out why it would do this on my end when nothing has changed.

**Comment by janstadt** [comment]
Only happens when i start with `network_mode: host`

Seems like its spinning up the slimserver but then it doesnt properly get it all the way set up to a state where the provider is enabled. When i try and enable the provider it throws that error. 

```
2024-04-11 13:32:47.817 ERROR (MainThread) [music_assistant] Error loading provider(instance) Slimproto (Squeezebox players)
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/music_assistant/server/providers/slimproto/__init__.py", line 253, in handle_async_init
    await self.slimproto.start()
  File "/usr/local/lib/python3.12/site-packages/aioslimproto/server.py", line 75, in start
    await self.cli.start()
  File "/usr/local/lib/python3.12/site-packages/aioslimproto/cli.py", line 176, in start
    await self._tcp_site.start()
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_runner.py", line 119, in start
    self._server = await loop.create_server(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 1572, in create_server
    raise OSError(err.errno, msg) from None
OSError: [Errno 98] error while attempting to bind on address ('0.0.0.0', 9000): address already in use

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/music_assistant/server/server.py", line 536, in load_provider
    await self.load_provider(prov_conf)
  File "/usr/local/lib/python3.12/site-packages/music_assistant/server/server.py", line 452, in load_provider
    provider = await prov_mod.setup(self, prov_manifest, conf)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/music_assistant/server/providers/slimproto/__init__.py", line 143, in setup
    await prov.handle_async_init()
  File "/usr/local/lib/python3.12/site-packages/music_assistant/server/providers/slimproto/__init__.py", line 255, in handle_async_init
    raise SetupFailedError(
music_assistant.common.models.errors.SetupFailedError: Unable to start the Slimproto server - is one of the required TCP ports already taken ?
```

**Comment by OzGav** [comment]
Yeah ok that error is saying that something else on your network is using port 9000. We had a report the other day from someone whose portainer was using that port? So have a think about what might be blocking

**Comment by janstadt** [comment]
Ok i figured this out. my reverse proxy (linuxserver SWAG) when running in host mode somehow uses up port 9000. When i use the default docker network, it doesnt. Super weird. I changed my SWAG config to host mode so that i didnt have to allowlist some trusted proxy config in homeassistant all the time. Quite weird its using ports only in host mode.

**Comment by janstadt** [comment]
Im still having issues with spotify and plex playing anything. I presume i have to wait for each provider to be fully synchronized, but i found a song that has been added to my library and i tried playing it on a few speakers and it wont play. It tries to start then stops. Logs look like this:


```
2024-04-11 20:42:38.340 VERBOSE (MainThread) [music_assistant.streams.media_stream] Input #0, ogg, from 'fd:':
2024-04-11 20:42:38.340 VERBOSE (MainThread) [music_assistant.streams.media_stream] Duration: N/A, start: 0.000000, bitrate: 320 kb/s
2024-04-11 20:42:38.340 VERBOSE (MainThread) [music_assistant.streams.media_stream] Stream #0:0: Audio: vorbis, 44100 Hz, stereo, fltp, 320 kb/s
2024-04-11 20:42:38.343 VERBOSE (MainThread) [music_assistant.streams.media_stream] Stream mapping:
2024-04-11 20:42:38.343 VERBOSE (MainThread) [music_assistant.streams.media_stream] Stream #0:0 -> #0:0 (vorbis (native) -> flac (native))
2024-04-11 20:42:38.343 VERBOSE (MainThread) [music_assistant.streams.media_stream] [Parsed_aresample_1 @ 0x55806921fa80] Cannot select channel layout for the link between filters Parsed_aresample_1 and format_out_0_0.
2024-04-11 20:42:38.343 VERBOSE (MainThread) [music_assistant.streams.media_stream] Error reinitializing filters!
2024-04-11 20:42:38.343 VERBOSE (MainThread) [music_assistant.streams.media_stream] Failed to inject frame into filter network: Invalid argument
2024-04-11 20:42:38.343 VERBOSE (MainThread) [music_assistant.streams.media_stream] Error while processing the decoded data for stream #0:0
2024-04-11 20:42:38.343 VERBOSE (MainThread) [music_assistant.streams.media_stream] Conversion failed!
2024-04-11 20:42:38.350 DEBUG (MainThread) [music_assistant.streams.media_stream] Process ffmpeg with PID 200 stopped with returncode 1
2024-04-11 20:42:38.350 DEBUG (MainThread) [music_assistant.streams.media_stream] stream finished (with code 1) for spotify--jfAEKTRa://track/5gCruy9T6XJcy4tSKf5em8
2024-04-11 20:42:38.350 DEBUG (MainThread) [music_assistant.streams.media_stream] Process ffmpeg with PID 200 stopped with returncode 1
2024-04-11 20:42:41.352 DEBUG (MainThread) [music_assistant.streams] Got GET request to /single/b8:27:eb:e4:f5:0b/f6cdf0782d17443b80616264b7383b9f.flac from 192.168.1.236
headers: <CIMultiDictProxy('Host': '192.168.1.68:8098', 'Connection': 'close', 'Accept': '*/*', 'Cache-Control': 'no-cache', 'User-Agent': 'VLC/3.0.9 LibVLC/3.0.9', 'Range': 'bytes=0-')>

2024-04-11 20:42:41.352 DEBUG (MainThread) [music_assistant.streams] Start serving audio stream for QueueItem spotify://track/3ycnLH0vm7D9K2QHA4cAnq to The Kitchen
2024-04-11 20:42:41.352 DEBUG (MainThread) [music_assistant.streams.media_stream] start media stream for: spotify--jfAEKTRa://track/3ycnLH0vm7D9K2QHA4cAnq
2024-04-11 20:42:41.354 DEBUG (MainThread) [music_assistant.streams.media_stream] Process ffmpeg started with PID 206
2024-04-11 20:42:42.351 VERBOSE (MainThread) [music_assistant.streams.media_stream] Input #0, ogg, from 'fd:':
2024-04-11 20:42:42.351 VERBOSE (MainThread) [music_assistant.streams.media_stream] Duration: N/A, start: 0.000000, bitrate: 320 kb/s
2024-04-11 20:42:42.351 VERBOSE (MainThread) [music_assistant.streams.media_stream] Stream #0:0: Audio: vorbis, 44100 Hz, stereo, fltp, 320 kb/s
2024-04-11 20:42:42.354 VERBOSE (MainThread) [music_assistant.streams.media_stream] Stream mapping:
2024-04-11 20:42:42.354 VERBOSE (MainThread) [music_assistant.streams.media_stream] Stream #0:0 -> #0:0 (vorbis (native) -> flac (native))
2024-04-11 20:42:42.354 VERBOSE (MainThread) [music_assistant.streams.media_stream] [Parsed_aresample_1 @ 0x55db8bc46a80] Cannot select channel layout for the link between filters Parsed_aresample_1 and format_out_0_0.
2024-04-11 20:42:42.354 VERBOSE (MainThread) [music_assistant.streams.media_stream] Error reinitializing filters!
2024-04-11 20:42:42.355 VERBOSE (MainThread) [music_assistant.streams.media_stream] Failed to inject frame into filter network: Invalid argument
2024-04-11 20:42:42.355 VERBOSE (MainThread) [music_assistant.streams.media_stream] Error while processing the decoded data for stream #0:0
2024-04-11 20:42:42.355 VERBOSE (MainThread) [music_assistant.streams.media_stream] Conversion failed!
2024-04-11 20:42:42.364 DEBUG (MainThread) [music_assistant.streams.media_stream] Process ffmpeg with PID 206 stopped with returncode 1
2024-04-11 20:42:42.364 DEBUG (MainThread) [music_assistant.streams.media_stream] stream finished (with code 1) for spotify--jfAEKTRa://track/3ycnLH0vm7D9K2QHA4cAnq
2024-04-11 20:42:42.364 DEBUG (MainThread) [music_assistant.streams.media_stream] Process ffmpeg with PID 206 stopped with returncode 1
```

**Comment by janstadt** [comment]
Looks like there is another hotfix for something sounding similar. will pull that down and try again.

**Comment by janstadt** [comment]
Play is working on 235 now, but repeat still doesnt work. I tested on a squeezebox and tried with and without flow mode enabled. 
https://streamable.com/mwoock link to a screen recording. Let me know if you want more logs.

**Comment by janstadt** [comment]
```
2024-04-11 21:06:19.391 DEBUG (MainThread) [music_assistant.streams] Got GET request to /single/b8:27:eb:e4:f5:0b/1a1f4bec4db040efa254b57d1a691079.flac from 192.168.1.236
headers: <CIMultiDictProxy('Host': '192.168.1.68:8097', 'Connection': 'close', 'Accept': '*/*', 'Cache-Control': 'no-cache', 'User-Agent': 'VLC/3.0.9 LibVLC/3.0.9', 'Range': 'bytes=0-')>

2024-04-11 21:06:19.391 DEBUG (MainThread) [music_assistant.streams] Start serving audio stream for QueueItem library://track/7100 to The Kitchen
2024-04-11 21:06:19.392 DEBUG (MainThread) [music_assistant.streams.media_stream] start media stream for: filesystem_local--P7JjijKM://track/brown-noise.mp3
2024-04-11 21:06:19.393 DEBUG (MainThread) [music_assistant.streams.media_stream] Process ffmpeg started with PID 935
2024-04-11 21:06:19.399 DEBUG (MainThread) [aioslimproto.client.The Kitchen] RESP received - Response received at player.
2024-04-11 21:06:19.399 DEBUG (MainThread) [aioslimproto.client.The Kitchen] send CODC for contenttype audio/flac: b'f????'
2024-04-11 21:06:19.542 VERBOSE (MainThread) [music_assistant.streams.media_stream] Input #0, mp3, from '/media/brown-noise.mp3':
2024-04-11 21:06:19.542 VERBOSE (MainThread) [music_assistant.streams.media_stream] Metadata:
2024-04-11 21:06:19.542 VERBOSE (MainThread) [music_assistant.streams.media_stream] title           : Brown Noise
2024-04-11 21:06:19.542 VERBOSE (MainThread) [music_assistant.streams.media_stream] album           : Varioius Artists
2024-04-11 21:06:19.542 VERBOSE (MainThread) [music_assistant.streams.media_stream] album_artist    : Various Artists
2024-04-11 21:06:19.542 VERBOSE (MainThread) [music_assistant.streams.media_stream] track           : 1
2024-04-11 21:06:19.542 VERBOSE (MainThread) [music_assistant.streams.media_stream] disc            : 1
2024-04-11 21:06:19.542 VERBOSE (MainThread) [music_assistant.streams.media_stream] artist          : Various Artists
2024-04-11 21:06:19.542 VERBOSE (MainThread) [music_assistant.streams.media_stream] lyrics-eng      :
2024-04-11 21:06:19.542 VERBOSE (MainThread) [music_assistant.streams.media_stream] date            : 2024
2024-04-11 21:06:19.542 VERBOSE (MainThread) [music_assistant.streams.media_stream] Duration: 00:00:09.55, start: 0.023021, bitrate: 67 kb/s
2024-04-11 21:06:19.542 VERBOSE (MainThread) [music_assistant.streams.media_stream] Stream #0:0: Audio: mp3, 48000 Hz, mono, fltp, 64 kb/s
2024-04-11 21:06:19.543 VERBOSE (MainThread) [music_assistant.streams.media_stream] Stream mapping:
2024-04-11 21:06:19.543 VERBOSE (MainThread) [music_assistant.streams.media_stream] Stream #0:0 -> #0:0 (mp3 (mp3float) -> flac (native))
2024-04-11 21:06:19.543 VERBOSE (MainThread) [music_assistant.streams.media_stream] Press [q] to stop, [?] for help
2024-04-11 21:06:19.547 VERBOSE (MainThread) [music_assistant.streams.media_stream] [flac @ 0x55b817010800] encoding as 24 bits-per-sample, more is considered experimental. Add -strict experimental if you want to encode more than 24 bits-per-sample
2024-04-11 21:06:19.547 VERBOSE (MainThread) [music_assistant.streams.media_stream] Output #0, flac, to 'pipe:':
2024-04-11 21:06:19.547 VERBOSE (MainThread) [music_assistant.streams.media_stream] Metadata:
2024-04-11 21:06:19.547 VERBOSE (MainThread) [music_assistant.streams.media_stream] title           : Brown Noise
2024-04-11 21:06:19.547 VERBOSE (MainThread) [music_assistant.streams.media_stream] album           : Varioius Artists
2024-04-11 21:06:19.547 VERBOSE (MainThread) [music_assistant.streams.media_stream] ALBUMARTIST     : Various Artists
2024-04-11 21:06:19.547 VERBOSE (MainThread) [music_assistant.streams.media_stream] TRACKNUMBER     : 1
2024-04-11 21:06:19.547 VERBOSE (MainThread) [music_assistant.streams.media_stream] DISCNUMBER      : 1
2024-04-11 21:06:19.547 VERBOSE (MainThread) [music_assistant.streams.media_stream] artist          : Various Artists
2024-04-11 21:06:19.547 VERBOSE (MainThread) [music_assistant.streams.media_stream] lyrics-eng      :
2024-04-11 21:06:19.547 VERBOSE (MainThread) [music_assistant.streams.media_stream] date            : 2024
2024-04-11 21:06:19.547 VERBOSE (MainThread) [music_assistant.streams.media_stream] encoder         : Lavf60.3.100
2024-04-11 21:06:19.548 VERBOSE (MainThread) [music_assistant.streams.media_stream] Stream #0:0: Audio: flac, 48000 Hz, stereo, s32 (24 bit), 128 kb/s
2024-04-11 21:06:19.548 VERBOSE (MainThread) [music_assistant.streams.media_stream] Metadata:
2024-04-11 21:06:19.548 VERBOSE (MainThread) [music_assistant.streams.media_stream] encoder         : Lavc60.3.100 flac
2024-04-11 21:06:19.610 DEBUG (MainThread) [aioslimproto.client.The Kitchen] STMl received - Buffer threshold reached.
2024-04-11 21:06:19.610 DEBUG (MainThread) [aioslimproto.client.The Kitchen] STMr received - resume confirmed.
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] [flac @ 0x55b817004b00] unable to rewrite FLAC header.
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] size=     955kB time=00:00:09.60 bitrate= 814.8kbits/s speed=46.9x
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] video:0kB audio:947kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.871899%
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] [Parsed_loudnorm_0 @ 0x55b81701d500]
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] {
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] "input_i" : "-25.24",
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] "input_tp" : "-1.61",
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] "input_lra" : "0.30",
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] "input_thresh" : "-35.24",
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] "output_i" : "-20.94",
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] "output_tp" : "-1.50",
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] "output_lra" : "1.00",
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] "output_thresh" : "-30.94",
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] "normalization_type" : "dynamic",
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] "target_offset" : "3.94"
2024-04-11 21:06:19.748 VERBOSE (MainThread) [music_assistant.streams.media_stream] }
2024-04-11 21:06:19.759 DEBUG (MainThread) [music_assistant.streams.media_stream] stream finished (with code 0) for filesystem_local--P7JjijKM://track/brown-noise.mp3 - seconds streamed: 9.6
2024-04-11 21:06:19.759 DEBUG (MainThread) [music_assistant.streams.media_stream] Loudness measurement for filesystem_local--P7JjijKM://track/brown-noise.mp3: LoudnessMeasurement(integrated=-25.24, true_peak=-1.61, lra=0.3, threshold=-35.24, target_offset=3.94)
2024-04-11 21:06:19.759 DEBUG (MainThread) [music_assistant.streams.media_stream] Process ffmpeg with PID 935 stopped with returncode 0
2024-04-11 21:06:19.770 DEBUG (MainThread) [aioslimproto.client.The Kitchen] No handler for dsco
2024-04-11 21:06:19.810 DEBUG (MainThread) [aioslimproto.client.The Kitchen] STMs received - playback of new track has started
2024-04-11 21:06:20.562 DEBUG (MainThread) [aioslimproto.client.The Kitchen] STMd received - decoder ready.
2024-04-11 21:06:21.356 DEBUG (MainThread) [aioslimproto.client.The Kitchen] play url (enqueue: True): http://192.168.1.68:8097/single/b8:27:eb:e4:f5:0b/1a1f4bec4db040efa254b57d1a691079.flac?ts=1712869581
2024-04-11 21:06:21.377 DEBUG (MainThread) [aioslimproto.client.The Kitchen] play url (enqueue: True): http://192.168.1.68:8097/single/b8:27:eb:e4:f5:0b/1a1f4bec4db040efa254b57d1a691079.flac?ts=1712869581
2024-04-11 21:06:30.177 DEBUG (MainThread) [aioslimproto.client.The Kitchen] STMu received - end of playback.
```

**Comment by marcelveldt** [comment]
Are you testing with an actual **hardware** squeezebox perhaps ?
Those have issues transitioning to the next track (so that includes repeat) - the best way to fix it is to enable "flow mode" in the player settings.

BTW: a new update is coming up with a bunch of buxfixes for streaming again - I spot some errors in your logs that are fixed today.

**Comment by OzGav** [comment]
@janstadt beta136 is released. Please check again

**Comment by janstadt** [comment]
Ok repeat is working now. I do notice one interesting thing. If i play this mp3 with the volume normalization setting on the mp3 sounds off. Like its pulsing or going in and out. If i turn off volume normalization it plays just fine.  Any ideas what would cause that?
[brown-noise.mp3.zip](https://github.com/music-assistant/hass-music-assistant/files/14971082/brown-noise.mp3.zip)

**Comment by janstadt** [comment]
D@mn, i take it back. It repeats for like 5 repetitions and then stops. Just stops. 

```
2024-04-14 14:39:40.672 DEBUG (MainThread) [music_assistant.streams] Got GET request to /single/00:00:00:00:00:00/baa9561338124cdab6b073c672b5c3ec.flac from 192.168.1.193
headers: <CIMultiDictProxy('Host': '192.168.1.68:8098', 'Connection': 'close', 'Accept': '*/*', 'Cache-Control': 'no-cache', 'User-Agent': 'VLC/3.0.9 LibVLC/3.0.9', 'Range': 'bytes=0-')>

2024-04-14 14:39:40.673 DEBUG (MainThread) [music_assistant.streams] Start serving audio stream for QueueItem library://track/7100 to The Deck
2024-04-14 14:39:40.673 DEBUG (MainThread) [music_assistant.audio.ffmpeg] starting ffmpeg with args: -hide_banner -loglevel info -nostats -ignore_unknown -protocol_whitelist file,hls,http,https,tcp,tls,crypto,pipe,data,fd,rtp,udp -readrate 1.2 -ac 2 -channel_layout stereo -ar 48000 -acodec pcm_s24le -f s24le -i - -af aresample=resampler=soxr:precision=28 -f flac -ar 48000 -ac 2 -
2024-04-14 14:39:40.674 DEBUG (MainThread) [music_assistant.audio.ffmpeg] Process ffmpeg started with PID 3787
2024-04-14 14:39:40.674 DEBUG (MainThread) [music_assistant.streams.media_stream] start media stream for: filesystem_local--P7JjijKM://track/brown-noise.mp3
2024-04-14 14:39:40.674 DEBUG (MainThread) [music_assistant.streams.media_stream] starting ffmpeg with args: -hide_banner -loglevel info -nostats -ignore_unknown -protocol_whitelist file,hls,http,https,tcp,tls,crypto,pipe,data,fd,rtp,udp -filter_threads 2 -i <FILE> -acodec pcm_s24le -f s24le -ar 48000 -ac 2 -
2024-04-14 14:39:40.675 DEBUG (MainThread) [music_assistant.streams.media_stream] Process ffmpeg started with PID 3788
2024-04-14 14:39:40.708 DEBUG (MainThread) [aioslimproto.client.The Deck] RESP received - Response received at player.
2024-04-14 14:39:40.708 DEBUG (MainThread) [aioslimproto.client.The Deck] send CODC for contenttype audio/flac: b'f????'
2024-04-14 14:39:40.708 DEBUG (MainThread) [aioslimproto.client.The Deck] STMc received - connected.
2024-04-14 14:39:45.498 DEBUG (MainThread) [aioslimproto.client.The Deck] STMs received - playback of new track has started
2024-04-14 14:39:45.626 DEBUG (MainThread) [music_assistant.streams.media_stream] Process ffmpeg with PID 3788 stopped with returncode 0
2024-04-14 14:39:45.626 DEBUG (MainThread) [music_assistant.streams.media_stream] stream finished (with code 0) for filesystem_local--P7JjijKM://track/brown-noise.mp3 - seconds streamed: 14.26571875
2024-04-14 14:39:46.463 DEBUG (MainThread) [music_assistant.webserver] Handling command music/item_by_uri
2024-04-14 14:39:46.630 DEBUG (MainThread) [aioslimproto.client.The Deck] play url (enqueue: True): http://192.168.1.68:8098/single/00:00:00:00:00:00/baa9561338124cdab6b073c672b5c3ec.flac?ts=1713105586
2024-04-14 14:39:46.637 DEBUG (MainThread) [music_assistant.webserver] Handling command music/item_by_uri
2024-04-14 14:39:46.838 DEBUG (MainThread) [aioslimproto.client.The Deck] play url (enqueue: True): http://192.168.1.68:8098/single/00:00:00:00:00:00/baa9561338124cdab6b073c672b5c3ec.flac?ts=1713105586
2024-04-14 14:39:48.766 DEBUG (MainThread) [music_assistant.audio.ffmpeg] Process ffmpeg with PID 3787 stopped with returncode 0
2024-04-14 14:39:48.766 DEBUG (MainThread) [aioslimproto.client.The Deck] No handler for dsco
2024-04-14 14:39:48.782 DEBUG (MainThread) [aioslimproto.client.The Deck] STMd received - decoder ready.
2024-04-14 14:39:48.782 DEBUG (MainThread) [aioslimproto.client.The Deck] play url (enqueue: False): http://192.168.1.68:8098/single/00:00:00:00:00:00/baa9561338124cdab6b073c672b5c3ec.flac?ts=1713105586
2024-04-14 14:39:48.783 DEBUG (MainThread) [aioslimproto.client.The Deck] STMf received - connection closed.
2024-04-14 14:39:48.823 DEBUG (MainThread) [aioslimproto.client.The Deck] STMc received - connected.
2024-04-14 14:39:48.866 DEBUG (MainThread) [music_assistant.streams] Got GET request to /single/00:00:00:00:00:00/baa9561338124cdab6b073c672b5c3ec.flac from 192.168.1.193
headers: <CIMultiDictProxy('Host': '192.168.1.68:8098', 'Connection': 'close', 'Accept': '*/*', 'Cache-Control': 'no-cache', 'User-Agent': 'VLC/3.0.9 LibVLC/3.0.9', 'Range': 'bytes=0-')>

2024-04-14 14:39:48.867 DEBUG (MainThread) [music_assistant.streams] Start serving audio stream for QueueItem library://track/7100 to The Deck
2024-04-14 14:39:48.867 DEBUG (MainThread) [music_assistant.audio.ffmpeg] starting ffmpeg with args: -hide_banner -loglevel info -nostats -ignore_unknown -protocol_whitelist file,hls,http,https,tcp,tls,crypto,pipe,data,fd,rtp,udp -readrate 1.2 -ac 2 -channel_layout stereo -ar 48000 -acodec pcm_s24le -f s24le -i - -af aresample=resampler=soxr:precision=28 -f flac -ar 48000 -ac 2 -
2024-04-14 14:39:48.868 DEBUG (MainThread) [music_assistant.audio.ffmpeg] Process ffmpeg started with PID 3820
2024-04-14 14:39:48.868 DEBUG (MainThread) [music_assistant.streams.media_stream] start media stream for: filesystem_local--P7JjijKM://track/brown-noise.mp3
2024-04-14 14:39:48.868 DEBUG (MainThread) [music_assistant.streams.media_stream] starting ffmpeg with args: -hide_banner -loglevel info -nostats -ignore_unknown -protocol_whitelist file,hls,http,https,tcp,tls,crypto,pipe,data,fd,rtp,udp -filter_threads 2 -i <FILE> -acodec pcm_s24le -f s24le -ar 48000 -ac 2 -
2024-04-14 14:39:48.868 DEBUG (MainThread) [music_assistant.streams.media_stream] Process ffmpeg started with PID 3821
2024-04-14 14:39:48.870 DEBUG (MainThread) [aioslimproto.client.The Deck] RESP received - Response received at player.
2024-04-14 14:39:48.870 DEBUG (MainThread) [aioslimproto.client.The Deck] send CODC for contenttype audio/flac: b'f????'
2024-04-14 14:39:53.796 DEBUG (MainThread) [music_assistant.streams.media_stream] Process ffmpeg with PID 3821 stopped with returncode 0
2024-04-14 14:39:53.796 DEBUG (MainThread) [music_assistant.streams.media_stream] stream finished (with code 0) for filesystem_local--P7JjijKM://track/brown-noise.mp3 - seconds streamed: 14.26571875
2024-04-14 14:39:54.446 DEBUG (MainThread) [music_assistant.webserver] Handling command music/item_by_uri
2024-04-14 14:39:55.428 DEBUG (MainThread) [aioslimproto.client.The Deck] STMs received - playback of new track has started
2024-04-14 14:39:56.966 DEBUG (MainThread) [music_assistant.audio.ffmpeg] Process ffmpeg with PID 3820 stopped with returncode 0
2024-04-14 14:39:56.966 DEBUG (MainThread) [aioslimproto.client.The Deck] No handler for dsco
2024-04-14 14:39:56.998 DEBUG (MainThread) [aioslimproto.client.The Deck] STMd received - decoder ready.
2024-04-14 14:40:04.434 DEBUG (MainThread) [aioslimproto.client.The Deck] STMu received - end of playback.
```

**Comment by marcelveldt** [comment]
I see you attached the file here, I will check it out.

BTW:
I notice one small issue in your log, which is the player ID being 00:00:00:00:00:00 that is super weird.
Is this a real squeezebox or a squeezelite based setup ?

**Comment by janstadt** [comment]
This is a squeezelite setup (picoreplayer).

**Comment by janstadt** [comment]
piCorePlayer v8.2.0 | www v00017 | linux 5.15.35-pcpCore-v7l (32) | piCore v13.2 | Squeezelite v1.9.9-1430-pCP

**Comment by marcelveldt** [comment]
> This is a squeezelite setup (picoreplayer).

Ah OK, There is a known issue with picoreplayer starting too soon so that it does not yet have the mac address, hence the 00 address. That will cause issues if you once decide to add another picoreplayer. In discord it was mentioned last week that there is a "network wait" option to delay the startup of squeezelite a bit on picoreplayer until the network is ready. You can also just use the -m option and specify the mac address yourself.

**Comment by marcelveldt** [comment]
I reproduced the issue with that brown noise file - looks like the volume normalization is completely tripping on the file.
I'll investigate more. Maybe I didnt account enough for the fact that the input fie is mono

**Comment by janstadt** [comment]
I'll take a look at the picore settings. Thanks for the heads up on the network wait stuff.

**Comment by marcelveldt** [comment]
Also reproduced the stop, that is just a race condition due to the file being this short. I will also look at fixing that

**Comment by janstadt** [comment]
Im curious if you would have a preference between airplay or squeezelite for playing music on these picoreplayers? I have shairport sync configured on them all but i noticed weird issues jumping between playing music to Airplay devices from spotify vs things like home assistant automations. I decided on going with just squeezelite but am wondering if all my issues would go away if i just used airplay? I know its an inferior protocol for playback and whatnot, but im not a huge audiophile and i think it would make my wife and kids' lives easier if they could just use airplay to play stuff. Unsure if that would really solve everything but who knows.

**Comment by janstadt** [comment]
Oh weird, just one of my devices are showing the 00: mac addresses. Even with the wait time modification. I'll need to look at that and see why the player is showing zeroes for its mac.

**Comment by marcelveldt** [comment]
squeezelite should work great - you just have an edge case with this short brown noise file playing on repeat.
A LOT of people are running squeezelite based players with the slimproto provider in MA just fine for their daily playback.

As for Airplay: that is also a great protocol and I actually prefer it over slimproto because it has great accurate sync being a timestamped protocol vs the crappy serverside compensation of slimproto.

Airplay is limited to redbook CD quality only (44.1/16bits) but that is sufficient for everyone.
Only on 50K speakers you may be able to tell the difference between higher sample rates but still I highly doubt that.

So, both are fine. Airplay is great, slimproto is also great.
I use Airplay a lot myself as it allows me to stream to a variation of speakers I have.

**Comment by marcelveldt** [comment]
Another update. A short track on single repeat was just bound to fail. The squeezelite player has a buffer of about 10 seconds and loads the clip (super fast) into memory, then asks the server for a next track but that has no idea that the player even started playing already because the request comes in only a few milliseconds after actually sending the request to start playback. Its a good old race condition. The MA server is optimized to allow players to buffer more for smoother playback while LMS tends to be a bit slower so the race condition trips not as easy there but it also does exist.

I have put in a bandaid in the code and in my testing its now working fine.

Then to the next issue. The volume normalization was not working correctly on this brown noise audio clip because the limiter would exceed the bounds (the clip has a super high dynamic range) - I have now optimized our code to auto detect this and dynamically adjust the target loudness level to account for more headroom. The file might play crappy the first time but then the measurement is received and it will use the adjusted values in the subsequent runs. 

So, long story short, the issues should be solved in the next update.

**Comment by janstadt** [comment]
Thanks for the info. I havent had an issue with LMS at all with the file so i wonder if im just getting lucky? I could maybe join the file to itself a bunch of times to make the clip longer. Would that further limit the race condition? For reference, i pulled the original wav file from https://moodist.app/ and then converted it to an mp3 after adding metadata to the file in order for it to play in MA.

**Comment by marcelveldt** [comment]
Yeah, its just a race condition of a few milliseconds so maybe LMS just loads the file quicker or feeds the audio slower or whatever. I handled it now on our side so should be good. But yeah, joining the file is actually even better.

I quickly joined it for you, see attachment.

**Comment by janstadt** [comment]
Thanks for the zip! I'll close this out when i test out the next release. Thanks again for all the help!

**Comment by OzGav** [comment]
@janstadt beta139 is released

**Comment by janstadt** [comment]
Thanks for the heads up. Replay is working longer than 5 replays now, but i still had to turn of Volume Normalization on the speakers in order for it to sound appropriate. If it is on, the volume spikes and muffles and isnt a constant stream of audio. It also wouldnt replay appropriately. This is on 138 though, i'll try and update and see if that helps

**Comment by janstadt** [comment]
I also notice that when i have repeat all on over airplay or squeezelite its not seamless even with crossfade one. Audible silence for a second or two and them it starts playing again.

**Comment by marcelveldt** [comment]
1) The first time you play a file like this (which is really bad recorded btw!) - it will have some difficulties to to the huge peaks vs low baseline but the subsequent runs of this file will run in lineair mode and should then run fine from now on. If it keeps giving issues, I can remaster that file for you.

2) repeat ALL will loop back the queue, it will stop the stream so it will never be continuous and I have no intention of changing that tbh as its a nightmare to support for all scenarios. repeat single (or just queued tracks) will not stop the output and keep going continuously.

3) you said that it seemed to ignore repeat while playing - was that a very short track again perhaps and only one track in the queue ? Also, was that with full repeat or single track ?

**Comment by janstadt** [comment]
Excellent information, thanks. I was not aware that was how repeat all vs single worked. I just tried repeat single with that file (with and without crossfade) and its still gapping with an audible break. If you think remastering that file would fix it, yeah that'd be awesome if you could do that for me. Much appreciated. The ignoring of repeat was on full repeat but i think it was a short track so that might be that race condition cropping up. The file you sent me that was twice as long doesnt have a problem repeating, it just 1) still playing oddly with normalization and 2) gapping on each replay when i have it on repeat single (and all but thats i suppose the same as repeat single if all i have is 1 song in the playlist).

**Comment by janstadt** [comment]
I've found quite a few different brown noise files online that seem to all exhibit that muffled pulsing sound where they just dont sound correct when going through volume normalization. Unsure about how to share these kinda files but i found some on archive dot org: https://shorturl.at/brtKL Tested out the 10hr one and its doing that odd playback as well.

**Comment by OzGav** [comment]
Yeah in testing I am also seeing unusual playback with my countdown test track. Repeat works fine but the sound level is off. If I go to the track view and press the preview button then it sounds fine.

[countdwn-23609.zip](https://github.com/music-assistant/hass-music-assistant/files/15070646/countdwn-23609.zip)

**Comment by marcelveldt** [comment]
LMS doesn't do EBU based volume normalization but just plain old replay gain, it can not be compared.

I have identified the ffmpeg loudnorm filter acting up on this brown noise file while in the first pass (dynamic mode) but it should be fixed once it has a measurement collecte din the db and do any second phase runs (so any time you play the track after the first run). At least in my case when I test with your file it sounds good and no more muffled pulsing sounds.

The other way I can think of is just detecting these sound effects files and completely disable volume normalization when we detect these. This can be as simple as adding something to the ID3 tags.

**Comment by marcelveldt** [comment]
Our entire enqueing logic is not really optimized for gapless repeat of short sound effects as we're built with music in mind. I will have another look if I can improve it without making the code too horrible.

Having gapless repeat for the entire queue (even with just one track in it) is something I'm not going to support as its much cleaner and maintainable to just restart the queue. Its also considered such an edge case of an edge case I don't think I should spend energy on it.

What should work is a continuous repeat of a single track but its a bit player dependent as we already figured out. MA does not only support slimproto (like LMS) but needs to support all modern protocols and player types so its a bit more of a challenge sometimes to satisfy everybody. For example we focus on having large buffers and allow players to buffer as much ahead as they want/can which prevents glitches/artifacts so the best possible playback of MUSIC is ensured.

That does not play well with super short audio clips on repeat where the player itself already filled its buffer before we can even think of feeding a next track.

Long story short, I will have another look at the repeat single track scenario.

Can you confirm that playing the brown noise file on next attempts has no more  muffled sound ?

**Comment by marcelveldt** [comment]
> Yeah in testing I am also seeing unusual playback with my countdown test track. Repeat works fine but the sound level is off. If I go to the track view and press the preview button then it sounds fine.

explain "sound level is off" - you mean the volume ?

**Comment by OzGav** [comment]
Yes  the sound level is low (compared to the preview playback) and it varies throughout the playback.

**Comment by janstadt** [comment]
> Our entire enqueing logic is not really optimized for gapless repeat of short sound effects as we're built with music in mind. I will have another look if I can improve it without making the code too horrible.
> 
> Having gapless repeat for the entire queue (even with just one track in it) is something I'm not going to support as its much cleaner and maintainable to just restart the queue. Its also considered such an edge case of an edge case I don't think I should spend energy on it.
> 
> What should work is a continuous, gapless repeat of a single track but its a bit player dependent as we already figured out. MA does not only support slimproto (like LMS) but needs to support all modern protocols and player types so its a bit more of a challenge sometimes to satisfy everybody. For example we focus on having large buffers and allow players to buffer as much ahead as they want/can which prevents glitches/artifacts so the best possible playback of MUSIC is ensured.
> 
> That does not play well with super short audio clips on repeat where the player itself already filled its buffer before we can even think of feeding a next track.
> 
> Long story short, I will have another look at the repeat single track scenario.
> 
> Can you confirm that playing the brown noise file on next attempts has no more muffled sound ?

Muffling is present no matter how many times the file has repeated.

**Comment by janstadt** [comment]
@marcelveldt I figure i'd ask here before submitting a new discussion. I have MA running and detecting my airplay speakers. If i cast music to one of the airplay speakers from my iphone, the track information (all information) doesnt seem to make it to MA (and therefore not to HA). The speakers in MA show off or idle even though they're playing audio from my phone. Is this by design? Is the goal for all music requests to originate from MA in order to get that kinda information into MA (and HA)? 

FWIW this same thing happens with LMS with airplay speakers, but i recently found the Shairtunes2 plugin for LMS which is a daapd fork. So what i do is on all my pcp players i disable shairtunes/airplay and just keep them as squeezelite players, but then with the Shairtunes2 plugin, it recognizes those squeezelite players and exposes them as Airplay devices. Now when i cast my music from my phone, the information is populated in LMS and HA which is what i want. Have you considered this approach to Airplay devices or is there something wrong with my setup that MA and HA arent getting audio metadata when i cast audio directly to Shairport sync on a pi? Any ideas would be great.

**Comment by OzGav** [comment]
If MA doesn’t initiate the stream then it doesn’t have control of the device and thus doesn’t know the state. The convoluted solution you have with LMS I don’t think has been considered, MA is not trying to be a LMS clone. What has been requested before and is being considered for the future is to make MA an Airplay target itself.

**Comment by janstadt** [comment]
That is effectively what Shairtunes2 does so that would be a nice future feature.

**Comment by marcelveldt** [comment]
> Muffling is present no matter how many times the file has repeated.

I think we have no other way than to just disable volume normalization on these effect files, the ffmpeg loudnorm filter clearly has issues with it while it works brilliant for all other content. What I observed in these files is that they're full of junk with high peaks and such, probably a result of the "noise". Now I need to figure a way how to determine this in an automatic way so loudnorm is disabled for effect files, while enabled for music.

What I can do I can do is fallback to a standard gain control (like LMS) when we detect a file with these values.

**Comment by janstadt** [comment]
Hey @marcelveldt. Any updates on this issue? Also, im curious if you had a chance to remaster that wav (or mp3) file so that it doesnt have all the junk in it? I've tested out disabling normalization and im stil hearing that muffling as @OzGav mentioned as well. If a remastered file is all it takes to get things running smoothly that should be enough for me.

**Comment by OzGav** [comment]
Sorry this will be low priority at the moment

**Comment by marcelveldt** [comment]
> Hey @marcelveldt. Any updates on this issue? Also, im curious if you had a chance to remaster that wav (or mp3) file so that it doesnt have all the junk in it? I've tested out disabling normalization and im stil hearing that muffling as @OzGav mentioned as well. If a remastered file is all it takes to get things running smoothly that should be enough for me.

Well, I have just accidentally discovered why this was failing and fixed it right on time for the 2.0 RC.
Still, the first time you play the effect it will still be dancing (as it is measuring the file) but then it will do a linear run in all subsequent runs. I just listened to it with headphones and now the linear adjusted volume leveled one sounds exactly the same as the one without volume normalizatton enabled.

Case closed ;-)

**Comment by janstadt** [comment]
Great news! Thanks @marcelveldt for the update!

**Comment by OzGav** [comment]
If you are satisfied then please close this one.

**Comment by janstadt** [comment]
I was hoping to test it out first but it says it'll be in a RC and not the latest Beta.

**Comment by OzGav** [comment]
Oh yep. So much going on. Please close when you are happy.

**Comment by OzGav** [comment]
@janstadt Have you tested this?

**Comment by janstadt** [comment]
Playback sounds much better. I am still unable to get gapless playback though unfortunately, even when selecting Repeat 1. I know that @marcelveldt mentioned that its player dependent, so i wonder if there are any secrets on making gapless playback for airplay devices.

**Comment by OzGav** [comment]
Have you got flow mode enabled?

**Comment by janstadt** [comment]
Yes. Flow Mode is enabled.

**Comment by OzGav** [comment]
If the original issue is resolved I would like to close this and you could enquire on Discord about the gapless repeat as I am not aware of anything that can help with that. Sorry.
